tm2/pkg/bft/consensus/replay_test.go

```diff
@@ -1161,14 +1166,34 @@ func TestHandshakeUpdatesValidators(t *testing.T) {
 	assert.Equal(t, newValAddr, expectValAddr)
 }
 
-// returns the vals on InitChain
-type initChainApp struct {
-	abci.BaseApplication
-	vals []abci.ValidatorUpdate
-}
+func TestHandshakeGenesisResponseDeliverTx(t *testing.T) {
+	t.Parallel()
 
-func (ica *initChainApp) InitChain(req abci.RequestInitChain) abci.ResponseInitChain {
-	return abci.ResponseInitChain{
-		Validators: ica.vals,
+	const numInitResponses = 42
+
+	app := &abci_testing.MockApplication{
+		InitChainFn: func(req abci.RequestInitChain) abci.ResponseInitChain {
+			return abci.ResponseInitChain{
+				TxResponses: make([]abci.ResponseDeliverTx, numInitResponses),
+			}
+		},
 	}
+	clientCreator := proxy.NewLocalClientCreator(app)
+
+	config, genesisFile := ResetConfig("handshake_test_")
+	defer os.RemoveAll(config.RootDir)
+	stateDB, state, store := makeStateAndStore(config, genesisFile, "v0.0.0-test")
+
+	// now start the app using the handshake - it should sync
+	genDoc, _ := sm.MakeGenesisDocFromFile(genesisFile)
+	handshaker := NewHandshaker(stateDB, state, store, genDoc)
+	proxyApp := appconn.NewAppConns(clientCreator)
+	require.NoError(t, proxyApp.Start(), "Error starting proxy app connections")
+	defer func() { require.NoError(t, proxyApp.Stop()) }()
```

Comment:
```suggestion
        t.Cleanup(func(){
            require.NoError(t, proxyApp.Stop())
        })
```

Reply:
done in 87fe7d8b6f2f145dc4ef2cecb6577d15ab40d3ba